clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy

```diff
@@ -99,15 +105,24 @@ abstract class AbstractEnableDisableKubernetesAtomicOperation implements AtomicO
         throw new KubernetesOperationException("Server group failed to reach a consistent state. This is likely a bug with Kubernetes itself.")
       }
     }
-    
+    def application
+    def clusterName
     if (replicationController) {
-      pods = credentials.apiAdaptor.getReplicationControllerPods(namespace, description.serverGroupName)
+      application = replicationController.metadata.labels.get("app")
```

Comment:
This relies on Spinnaker's labeling scheme, which is what we are trying to avoid.